migration/sql-files/041-unique-area-name-create-new-area.sql

```diff
@@ -0,0 +1,80 @@
+------ You can't allow the same area name and the same ancestry inside a space
+
+ALTER TABLE areas ADD CONSTRAINT areas_name_space_id_path_unique UNIQUE(space_id,name,path);
+
+------  For existing spaces in production, which dont have a default area, create one.
+--
+-- 1. Get all spaces which have an area under it with the same name.
+-- 2. Get all spaces not in (1)
+-- 3. insert an 'area' for all such spaces in (2)
+
+INSERT INTO areas
+            (created_at,
+             updated_at,
+             name,
+             space_id)
+SELECT current_timestamp,
+       current_timestamp,
+       name,
+       id
+FROM   spaces
+WHERE  id NOT IN (SELECT s.id
+                  FROM   spaces AS s
+                         INNER JOIN areas AS a
+                                 ON s.name = a.name
+                                    AND s.id = a.space_id);  
+
+
+----- for all other existing areas in production, move them under the default 'root' area.
+
+
+-- Get Root area for a space
+                           
+CREATE OR REPLACE FUNCTION GetRootArea(s_id uuid,OUT root_id uuid) AS $$ BEGIN
+     select id from areas 
+          where name in ( SELECT name as space_name
+          from spaces 
+              where id=s_id )
+                  and space_id =s_id 
+                           into root_id;
+END; $$ LANGUAGE plpgsql ;
+
+-- Convert Text to Ltree , use standard library FUNCTION?
+
+CREATE OR REPLACE FUNCTION TextToLtreeNode(u text, OUT node ltree) AS $$ BEGIN
```

Comment:
We already have functions dealing with this [here](https://github.com/almighty/almighty-core/blob/master/migration/sql-files/035-wit-to-use-uuid.sql#L40):

```sql
CREATE OR REPLACE FUNCTION UUIDToLtreeNode(u uuid, OUT node ltree) AS $$ BEGIN
-- Converts a UUID value into a value usable inside an Ltree 
    SELECT replace(u::text, '-', '_') INTO node;
END; $$ LANGUAGE plpgsql;

CREATE OR REPLACE FUNCTION LtreeNodeToUUID(node ltree, OUT u uuid) AS $$ BEGIN
-- Converts an Ltree node into a UUID value 
    SELECT replace(node::text, '_', '-') INTO u;
END; $$ LANGUAGE plpgsql;
```

Can you use them?